incubator/hnc/pkg/reconcilers/hierarchy_config.go

```diff
@@ -145,6 +156,70 @@ func (r *HierarchyConfigReconciler) reconcile(ctx context.Context, log logr.Logg
 	return r.updateObjects(ctx, log, nm)
 }
 
+// syncDeleting returns true if the namespace or the HC instance is being deleted.
+func (r *HierarchyConfigReconciler) syncDeleting(ctx context.Context, log logr.Logger, inst *api.HierarchyConfiguration, nsInst *corev1.Namespace) (bool, error) {
+	r.Forest.Lock()
```

Comment:
Normally, this would seem quite dangerous (eg no `defer unlock`) but I think once we simplify this code we'll be able to put it back. Can you put a TODO here to add a defer unlock?